CAIPs/caip-x.md

```diff
@@ -0,0 +1,339 @@
+---
+caip: XX
+title: JSON-RPC Provider Lifecycle Methods for Session Management
+author: [Alex Donesky] (@adonesky1)
+discussions-to: TBD
+status: Draft
+type: Standard
+created: 2024-06-07
+requires: 25, 217
+---
+
+## Simple Summary
+
+CAIP-XX introduces new "lifecycle" methods and a new event for managing permissions within an existing CAIP-25 session, allowing for addition, revocation, and retrieval of permissions. These methods provide an alternative to session management via `sessionId`s, allowing `sessionId`s to be optional for CAIP-25.
+
+## Abstract
+
+This proposal aims to extend the CAIP-25 standard by defining new JSON-RPC methods for managing the lifecycle of permissions within a session. These methods allow dapps and wallets to dynamically adjust permissions, providing more granular control and better user experience. Additionally, it allows for session management without mandatory sessionIds, offering more flexibility in handling sessions.
+
+## Motivation
+
+The motivation behind this proposal is to enhance the flexibility of CAIP-25 by enabling management of session permissions without sessionIds which don't map well to extension based wallet's dapp connections and could therefore add constraints and burdens to existing flows. The proposed methods provide an intuitive way to add, revoke, and retrieve permissions within an existing session, simplifying the management of session lifecycles.
+
+### Use Case Scenarios
+
+1. **Dapp Initiated Adding Permissions To an Existing Session:**
+
+   - **Current Method:** Call `provider_authorize` again with an existing session identifier and new scopes/methods to add. This is actually somewhat ambiguous in CAIP-25 where it’s unclear if incremental permissions adds should include the full object or just scopes to add: "This object gets updated, extended, closed, etc. by successive calls and notifications, each tagged by this identifier."
+   - **Proposed Method:** Use `provider_augment` to add new permissions to the existing session. Alternatively, this could be achieved with more specific language indicating that a subsequent `provider_authorize` call with new scopes/permissions could simply add to an existing session. If we pursue this latter approach one flow that will need clarity is what happens if a subsequent `provider_authorize` request includes `requiredScopes` which the respondent does not authorize. In this case I would expect that this does not revoke the existing session but rather only the new request is rejected.
```

Comment:
that breaks semantics of required, tho? as-is, response can keep changing and wallet can keep responding with the same exact response to "reject changes"

Reply:
Not sure I follow 🤔

Reply:
I guess my preference for the request to only require the new "permissions" is that the dapp doesn't need to store or keep track of permissions that it already has. 

[At the same time part of me is leaning back towards keeping everything atomic](https://github.com/ChainAgnostic/CAIPs/pull/285#issuecomment-2198782292) and just saying either `provider_authorize`/`wallet_createSession` or `provider_augment`/`wallet_updateSession` should just request full object every time to change session scopes.